scripts/src/java/org/oppia/android/scripts/xml/TextViewStyleCheck.kt

```diff
@@ -0,0 +1,289 @@
+
+    for (i in 0 until textViewNodes.length) {
+      val element = textViewNodes.item(i) as Element
+      validateTextViewElement(element, relativePath)
+    }
+  }
+
+  private fun validateTextViewElement(element: Element, filePath: String) {
+    val lineNumber = element.getAttribute(LINE_NUMBER_ATTRIBUTE).toInt().minus(1).toString()
+    val styleAttribute = element.attributes.getNamedItem("style")?.nodeValue
+    val idAttribute = element.attributes.getNamedItem("android:id")?.nodeValue ?: "NO ID"
+
+    if (idAttribute in attributeIds) return
+
+    if (styleAttribute.isNullOrBlank()) {
+      styleValidationIssues.add(
+        "ERROR: Missing style attribute in file: $filePath, line $lineNumber."
+      )
+    }
+
+    checkForLegacyDirectionality(element, filePath, lineNumber)
+  }
+
+  private fun checkForLegacyDirectionality(
+    element: Element,
+    filePath: String,
+    lineNumber: String
+  ) {
+    val legacyAttributes = listOf(
+      "android:paddingLeft",
+      "android:paddingRight",
+      "android:layout_marginLeft",
+      "android:layout_marginRight",
+      "android:layout_alignParentLeft",
+      "android:layout_alignParentRight",
+      "android:layout_toLeftOf",
+      "android:layout_toRightOf"
+    )
+
+    if (legacyAttributes.any { element.hasAttribute(it) }) {
+      directionalityWarnings.add(
+        "WARNING: Hardcoded left/right attribute in file: $filePath, line $lineNumber. " +
+          "Consider using start/end."
+      )
+    }
+  }
+
+  private fun printResults() {
+    directionalityWarnings.forEach { println(it) }
+
+    if (styleValidationIssues.isNotEmpty()) {
+      styleValidationIssues.forEach { println(it) }
+      throw Exception("TEXTVIEW STYLE CHECK FAILED")
+    } else if (directionalityWarnings.isEmpty()) {
+      println("TEXTVIEW STYLE CHECK PASSED")
+    }
+  }
+
+  private fun readXMLWithLineNumbers(inputStream: FileInputStream, lineNumAttribName: String):
+    Document {
+      val doc: Document
+      val parser: SAXParser
+      try {
+        val factory = SAXParserFactory.newInstance()
+        parser = factory.newSAXParser()
+        val docBuilderFactory = DocumentBuilderFactory.newInstance()
+        val docBuilder = docBuilderFactory.newDocumentBuilder()
+        doc = docBuilder.newDocument()
+      } catch (e: Exception) {
+        throw RuntimeException("Can't create SAX parser / DOM builder.", e)
+      }
+
+      val elementStack = Stack<Element>()
+      val textBuffer = StringBuilder()
+
+      val handler = object : DefaultHandler() {
+        private lateinit var locator: Locator
+
+        override fun setDocumentLocator(locator: Locator) {
+          this.locator = locator
+        }
+
+        override fun startElement(
+          uri: String,
+          localName: String,
+          qName: String,
+          attributes: Attributes
+        ) {
+          addTextIfNeeded()
+          val el = doc.createElement(qName)
```

Comment:
```suggestion
          val element = document.createElement(qName)
```

Suggestions for a more descriptive names.